### PR TITLE
feat: subscribe to flag notifications

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

What Neil wants he gets :)

I might want to see notifications about not just flags I created but specific important (to me) flags

## Changes

* Adds ability to add a subscription trigger that is "on change" and not "recurring"
* Adds a button to subscribe to flags
* Adds changes by other people for subscribed flags in the notification centre

subscription-to-flag

## TODO

still draft because...

- [ ] check this hasn't broken subscriptions
- [x] don't include changes from before subscribing to the flag
   This is tricky as a set with django ORM because there's no foreign key between the subscription and activity log. Let's not do it now. We should push into a `Notifications` table as we write to the activity log so we don't have to do this filtering at query time. See https://github.com/PostHog/meta/pull/62
- [x] show the subscribe button only for customers with subscriptions feature

## How did you test this code?

adding developer tests and running it in the UI

### CODE REVIEW

File: ee/tasks/subscriptions/__init__.py [hunk 1]
```diff
@@ -67,10 +67,12 @@ def _deliver_subscription_report(
         except Exception as e:
             incr("subscription_slack_send_failure")
             logger.error(e)
+    elif subscription.target_type == "in_app_notification":
+        incr("subscription_in_app_notification_success")
     else:
         raise NotImplementedError(f"{subscription.target_type} is not supported")
 
-    if not is_new_subscription_target:
+    if not is_new_subscription_target and subscription.target_type != "in_app_notification":
```
Comment (mariusandra) [inline ee/tasks/subscriptions/__init__.py:75]: Is this the right thing to compare to? Somehow I'd `and subscription.frequency != 'on_change'` feels more correct for checking to set the next date or not...?

**Review by mariusandra — APPROVED** [review]
Feels fine to merge this, though the water's not very shallow here... 😅

1. It feels like the API can accept values of any combo (e.g. daily flags, dashboard on change). Could it throw when you want to subscribe to something that doesn't make sense? What even makes sense?

2. In the todo: "[ ]  check this hasn't broken subscriptions"... -> Paul, did you check if you broke subscriptions? (I didn't 😅)

3. I'm worried the next 10 feature requests you'll answer are "can I subscribe to $thing". It might be worth just taking the plunge and building some kind of [notification store](https://github.com/PostHog/meta/pull/62/files) and getting the architecture right before doing all those one-off integrations :).

**Review by benjackwhite — COMMENTED** [review]
Took a brief pass through the code and nothing stood out as incorrect from an implementation perspective.

I do however question if we are potentially misusing the Subscription concept here for this. Are you actually `subscribing` to something in this case or is it rather you are creating custom filters for your own activity history feed?

I sort of agree with Marius that the next N requests will be "why can't I subscribe to changes on an Insight or Dashboard". Or - "I would also like an email subscription summarizing what things changed". What would this all look like? A subscription of subscriptions? 

Definitely like the effort put in here but I sort of agree with Marius that perhaps it makes sense to be slightly less YOLO with this feature and instead build it properly following something like the notifications RFC. At the very least leaving the Subscriptions as "Recurring subscriptions (i.e. timers)" and separating this out into some other model...

**Comment by pauldambra** [comment]
Hmm "makes sense to be slightly less YOLO with this feature "

I did find myself sending "that subscription is a trigger"

thirded

**Pull request closed without merging**